v3/src/components/graph/utilities/graph-utils.ts

```diff
@@ -275,3 +293,220 @@ export function computeSlopeAndIntercept(xAxis?: IAxisModel, yAxis?: IAxisModel)
+      sumOfProductDiffs: 0
+    }
+    let tSumDiffsX = 0
+    let tSumDiffsY = 0
+
+    // Under certain circumstances (adding new case) an empty value can sneak in here. Filter out.
+    iCoordPairs = iCoordPairs.filter((iPair: any) => {
+      return !(!iPair.x || !iPair.y)
+    })
+   iCoordPairs.forEach(function (iPair: any) {
+     if (isFinite(iPair.x) && isFinite(iPair.y)) {
+       tResult.count += 1
+       tResult.xSum += iPair.x
+       tResult.xSumOfSquares += (iPair.x * iPair.x)
+       tResult.ySum += iPair.y
+       tResult.ySumOfSquares += (iPair.y * iPair.y)
+     }
+   })
+   if (tResult.count > 0) {
+     tResult.xMean = tResult.xSum / tResult.count
+     tResult.yMean = tResult.ySum / tResult.count
+     iCoordPairs.forEach((iPair: any) => {
+       let tDiffX = 0
+       let tDiffY = 0
+       if (isFinite(iPair.x) && isFinite(iPair.y)) {
+         tResult.sumOfProductDiffs += (iPair.x - tResult.xMean) * (iPair.y - tResult.yMean)
+         tDiffX = iPair.x - tResult.xMean
+         tResult.xSumSquaredDeviations += tDiffX * tDiffX
+         tSumDiffsX += tDiffX
+         tDiffY = iPair.y - tResult.yMean
+         tResult.ySumSquaredDeviations += tDiffY * tDiffY
+         tSumDiffsY += tDiffY
+       }
+     })
+     // Subtract a correction factor for roundoff error.
+     // See Numeric Recipes in C, section 14.1 for details.
+     tResult.xSumSquaredDeviations -= (tSumDiffsX * tSumDiffsX / tResult.count)
+     tResult.ySumSquaredDeviations -= (tSumDiffsY * tSumDiffsY / tResult.count)
+   }
+   return tResult
+ }
+
+ const t_quantile_at_0975_for_df = [
+  [1, 12.7062],
+  [2, 4.30265],
+  [3, 3.18245],
+  [4, 2.77645],
+  [5, 2.57058],
+  [6, 2.44691],
+  [7, 2.36462],
+  [8, 2.306],
+  [9, 2.26216],
+  [10, 2.22814],
+  [11, 2.20099],
+  [12, 2.17881],
+  [13, 2.16037],
+  [14, 2.14479],
+  [15, 2.13145],
+  [16, 2.11991],
+  [17, 2.10982],
+  [18, 2.10092],
+  [19, 2.09302],
+  [20, 2.08596],
+  [21, 2.07961],
+  [22, 2.07387],
+  [23, 2.06866],
+  [24, 2.0639],
+  [25, 2.05954],
+  [26, 2.05553],
+  [27, 2.05183],
+  [28, 2.04841],
+  [29, 2.04523],
+  [30, 2.04227],
+  [40, 2.02108],
+  [50, 2.00856],
+  [60, 2.0003],
+  [70, 1.99444],
+  [80, 1.99006],
+  [90, 1.98667],
+  [100, 1.98397],
+  [200, 1.9719],
+  [500, 1.96472],
+  [1000, 1.96234],
+  [2000, 1.96115],
+  [10000, 1.9602],
+  [100000, 1.95999]
+]
+
+ export const tAt0975ForDf = (iDf: number) => {
+  const foundIndex = t_quantile_at_0975_for_df.findIndex((iPair: any) => iPair[0] > iDf)
+  return foundIndex <= 0 ? 1.96 : t_quantile_at_0975_for_df[foundIndex - 1][1]
+}
+export interface ISlopeIntercept {
+  count: number | null
+  intercept: number | null
+  mse: number | null // mean squared error
+  rSquared: number | null
+  sdResiduals: number | null
+  slope: number | null
+  sse: number // sum of squared errors
+  sumSquaresResiduals?: number | null
+  xMean: number | null
+  xSumSquaredDeviations: number | null
+  yMean?: number | null
```

Comment:
Is there a reason to use `null` here? My inclination would have been to make them all optional properties so they get initialized to `undefined` automatically.